src/api/test/unit/code_quality_test.rb

```diff
@@ -87,7 +87,7 @@ def setup
       'PersonController#internal_register'                                      => 112.01,
       'Package#find_changed_issues'                                             => 93.74,
       'Package#close_requests'                                                  => 84.82,
-      'Flag#default_status'                                                     => 90.56,
+      'Flag#compute_status'                                                     => 145.55,
```

Comment:
OK. I thought you found something else. I thought long about how to simplify the many if's and unless expressions. I used the tricks from the former code. If I find a quiet minute maybe I check if the monster can be simplified somehow :)